Need to parse server.xml for variables to resolve configure-arquillian http port

Currently we define ports as variables in the `server.xml` file (sample below), they are being resolved in bootstrap properties. Perhaps that's why running the `configure-arquilliam`  goal in the new LMP still requires the bootstrap property for the port number; otherwise it fails error: 

```
[INFO] --- liberty-maven-plugin:3.1:configure-arquillian (default-cli) @ guide-arquillian-managed ---
[INFO] CWWKM2102I: Using artifact based assembly archive : io.openliberty:openliberty-kernel:null:19.0.0.12:zip.
[INFO] CWWKM2102I: Using installDirectory : /Users/evelinecai/OLG/testing/4Qtasks/dev-mode/guide-arquillian-managed/finish/target/liberty/wlp.
[INFO] CWWKM2102I: Using serverName : defaultServer.
[INFO] CWWKM2102I: Using serverDirectory : /Users/evelinecai/OLG/testing/4Qtasks/dev-mode/guide-arquillian-managed/finish/target/liberty/wlp/usr/servers/defaultServer.
[INFO] Automatically detected the Arquillian Liberty Managed container at the following coordinates: io.openliberty.arquillian:arquillian-liberty-managed.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.035 s
[INFO] Finished at: 2019-12-13T11:04:16-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.openliberty.tools:liberty-maven-plugin:3.1:configure-arquillian (default-cli) on project guide-arquillian-managed: Error configuring Arquillian.: Unable to find variable "default.http.port" in bootstrap properties. -> [Help 1]
[ERROR] 
```

server.xml sample: 
```
<server description="new server">
    <featureManager>
        <feature>jaxrs-2.1</feature>
        <feature>jsonp-1.1</feature>
        <feature>cdi-2.0</feature>
        <feature>mpConfig-1.3</feature>
        <!--Enable the following features to run tests with Arquillian managed container-->
        <!-- tag::localConnector[] -->
        <feature>localConnector-1.0</feature>
        <!-- end::localConnector[] -->
        <!-- tag::Servlet[] -->
        <feature>servlet-4.0</feature>
        <!-- end::Servlet[] -->
    </featureManager>
    <variable name="default.http.port" defaultValue="9080" />
    <variable name="default.https.port" defaultValue="9443" />
    <httpEndpoint id="defaultHttpEndpoint" httpPort="${default.http.port}"
        httpsPort="${default.https.port}" />
    <webApplication location="guide-arquillian-managed.war"
        context-root="/" />
</server>
```

Discussed with @chyt. He will make the appropriate fix.

--- a/liberty-maven-plugin/pom.xml
+++ b/liberty-maven-plugin/pom.xml
@@ -84,7 +84,7 @@
         <dependency>
             <groupId>io.openliberty.tools</groupId>
             <artifactId>ci.common</artifactId>
-            <version>1.8.4</version>
+            <version>1.8.5-SNAPSHOT</version>
         </dependency>
         <dependency>
             <groupId>org.twdata.maven</groupId>

--- a/liberty-maven-plugin/src/it/arquillian-tests/pom.xml
+++ b/liberty-maven-plugin/src/it/arquillian-tests/pom.xml
@@ -67,7 +67,10 @@
         <warContext>myLibertyApp</warContext>
         <package.file>${project.build.directory}/${app.name}.zip</package.file>
         <packaging.type>usr</packaging.type>
-
+        
+        <liberty.var.default.http.port>9080</liberty.var.default.http.port>
+        <liberty.var.default.https.port>9443</liberty.var.default.https.port>
+        
         <!-- Test properties -->
         <arquillian.managed.artifact>arquillian-liberty-managed</arquillian.managed.artifact>
         <arquillian.managed.version>1.0.6</arquillian.managed.version>
@@ -174,8 +177,6 @@
                     <packageName>WebsocketServerPackage</packageName>
                     <include>runnable</include>
                     <bootstrapProperties>
-                        <default.http.port>9080</default.http.port>
-                        <default.https.port>9443</default.https.port>
                         <build.directory>${project.build.directory}</build.directory>
                     </bootstrapProperties>
                     <features>

--- a/liberty-maven-plugin/src/it/arquillian-tests/src/main/resources/server.xml
+++ b/liberty-maven-plugin/src/it/arquillian-tests/src/main/resources/server.xml
@@ -9,7 +9,7 @@
 
     <!-- To access this server from a remote client add a host attribute to
         the following element, e.g. host="*" -->
-    <httpEndpoint id="defaultHttpEndpoint" httpPort="9080"
-        httpsPort="9443" />
+    <httpEndpoint id="defaultHttpEndpoint" httpPort="${default.http.port}"
+        httpsPort="${default.https.port}" />
         
 </server>

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/extensions/arquillian/ConfigureArquillianMojo.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/extensions/arquillian/ConfigureArquillianMojo.java
@@ -1,5 +1,5 @@
 /**
- * (C) Copyright IBM Corporation 2017, 2018.
+ * (C) Copyright IBM Corporation 2017, 2020.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -148,8 +148,9 @@ public class ConfigureArquillianMojo extends BasicSupport {
     private int getHttpPort() throws FileNotFoundException, XPathExpressionException, IOException,
             ParserConfigurationException, SAXException, ArquillianConfigurationException {
         File serverXML = new File(serverDirectory + "/server.xml");
+        File configVariableXML = new File(serverDirectory + "/configDropins/overrides/liberty-plugin-variable-config.xml");
         File bootstrapProperties = new File(serverDirectory + "/bootstrap.properties");
-        return HttpPortUtil.getHttpPort(serverXML, bootstrapProperties);
+        return HttpPortUtil.getHttpPort(serverXML, bootstrapProperties, configVariableXML);
     }
 
 }

